src/frame/vulkan.rs

```diff
@@ -32,91 +32,109 @@ pub struct Vulkan {
 impl Vulkan {
     pub fn new() -> Result<Self, Box<dyn Error>> {
         let app_name = CString::new("wluma")?;
-        let app_info = vk::ApplicationInfo::builder()
-            .application_name(&app_name)
-            .application_version(WLUMA_VERSION)
-            .engine_name(&app_name)
-            .engine_version(WLUMA_VERSION)
-            .api_version(VULKAN_VERSION);
+        let app_info = vk::ApplicationInfo {
+            p_application_name: app_name.as_ptr(),
+            application_version: WLUMA_VERSION,
+            p_engine_name: app_name.as_ptr(),
+            engine_version: WLUMA_VERSION,
+            api_version: VULKAN_VERSION,
+            ..Default::default()
+        };
 
         let instance_extensions = &[
-            vk::KhrExternalMemoryCapabilitiesFn::name().as_ptr(),
-            vk::KhrGetPhysicalDeviceProperties2Fn::name().as_ptr(),
+            CStr::from_bytes_with_nul(b"vkGetPhysicalDeviceProperties2\0")
+                .unwrap()
+                .as_ptr(),
         ];
 
         let entry = Entry::linked();
 
-        let create_info = vk::InstanceCreateInfo::builder()
-            .application_info(&app_info)
-            .enabled_extension_names(instance_extensions);
+        let create_info = vk::InstanceCreateInfo {
+            p_application_info: &app_info,
+            enabled_extension_count: instance_extensions.len() as u32,
+            pp_enabled_extension_names: instance_extensions.as_ptr(),
+            ..Default::default()
+        };
```

Comment:
From the quick glance, I think I prefer the changes made to this file in https://github.com/maximbaz/wluma/pull/86#issuecomment-2320923641 , except I like your solution to the error types, not to introduce a ton of `.expect()` calls. Would you like to try to converge your work and the work in https://github.com/maximbaz/wluma/pull/86#issuecomment-2320923641 ?